### PR TITLE
Zscaler - rate limit bug fix

### PR DESCRIPTION
**Pull request by guykeller**
## Status
Ready

## Description
Fixed a bug in Zscaler which caused too many requests (3 or more) made in a short interval to fail.
After the fix, tested up to 20 requests and they did not fail.
Added a test for the bug in the playbook as well.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by ronykoz** [comment]
fixes https://github.com/demisto/etc/issues/15722

**Comment by ronykoz** [comment]
Remove all the non-Zscaler related files

File: Integrations/integration-Zscaler.yml [hunk 1]
```diff
@@ -1028,6 +1037,6 @@ script:
       type: string
     description: Returns Zscaler default whitelist
   runonce: false
-releaseNotes: "Added option to disable proxy"
+releaseNotes: "Fixed rate limit error bug - now several requests in short interval won't fail. Added option to disable proxy"
```
Comment (ronykoz) [inline Integrations/integration-Zscaler.yml]:
```suggestion
releaseNotes: "Fixed rate limit error bug - now several requests in short interval will cause retry Added option to disable proxy"
```

File: Integrations/integration-Zscaler.yml [hunk 2]
```diff
@@ -84,7 +88,12 @@ script:
                 headers=headers
             )
             if res.status_code not in (200, 204):
-                raise Exception('Your request failed with the following error: ' + ERROR_CODES_DICT[res.status_code])
+                if res.status_code == EXCEEDED_RATE_LIMIT_STATUS_CODE and num_of_seconds_to_wait <= MAX_SECONDS_TO_WAIT:
+                    time.sleep(num_of_seconds_to_wait)
+                    #time.sleep(num_of_seconds_to_wait)
```
Comment (ronykoz) [inline Integrations/integration-Zscaler.yml]: remove

File: Integrations/integration-Zscaler.yml [hunk 3]
```diff
@@ -53,6 +54,9 @@ script:
     DEFAULT_HEADERS = {
         'content-type': 'application/json'
     }
+    EXCEEDED_RATE_LIMIT_STATUS_CODE = 429
+    MAX_SECONDS_TO_WAIT = 15
+    SLEEP_TIME_BEFORE_RETRY = 2
```
Comment (ronykoz) [inline Integrations/integration-Zscaler.yml]: remove

**Comment by Itay4** [comment]
Did you manage get the rate limit error be raised on our instance in the first place?

**Comment by ronykoz** [comment]
@itay4 yes he did

File: Integrations/integration-Zscaler.yml [hunk 4]
```diff
@@ -66,14 +70,17 @@ script:
...
 
     ''' HANDLE PROXY '''
+    proxies = {}
     if not PROXY:
         del os.environ['HTTP_PROXY']
         del os.environ['HTTPS_PROXY']
         del os.environ['http_proxy']
         del os.environ['https_proxy']
+    else:
+        proxies = {'http_proxy': os.environ['http_proxy'], 'https_proxy': os.environ['https_proxy']}
```
Comment (Itay4) [inline Integrations/integration-Zscaler.yml]: you are not doing anything with this..

**Comment by guykeller** [comment]
Closing this PR and opening a new one as it would be more cost effective than reverting all irrelevant changes

**Pull request closed without merging**